spring-cloud-stream/src/main/java/org/springframework/cloud/stream/converter/TupleJsonMessageConverter.java

```diff
@@ -79,4 +74,16 @@ public Object convertFromInternal(Message<?> message, Class<?> targetClass, Obje
 		return json;
 	}
 
+	@Override
+	public Object convertFromInternal(Message<?> message, Class<?> targetClass, Object conversionHint) {
+		String source;
+		if (message.getPayload() instanceof byte[]) {
+			source = new String((byte[]) message.getPayload());
+		}
+		else {
```

Comment:
`convertFromInternal` is supposed to do the conversion from the serialized/wire format - which can be either `String` or `byte[]`. I believe we should assume in general that messages bearing a `contentType` header should be either one of the two (perhaps with the sole exception of already deserialized objects, which can have the `application/java-object;type=xyz` header, which is sort of an anomaly).